### PR TITLE
[MODORDERS-1078] Avoid onSuccess terminal operators in the middle of executing chain

### PR DESCRIPTION
**Pull request by SerhiiNosko**
## Purpose
The handler used in onSuccess was called async and the method was not waiting for it to return. But also the API says:
WARNING: this is a terminal operation. If several handlers are registered, there is no guarantee that they will be invoked in order of registration.

We need to identify places where usage of onSuccess was incorrect and replace to use of map or compose methods
https://folio-org.atlassian.net/browse/MODORDERS-1078

### CODE REVIEW

File: src/main/java/org/folio/service/pieces/flows/update/PieceUpdateFlowInventoryManager.java [hunk 1]
```diff
@@ -52,10 +52,13 @@ public Future<Void> processInventory(PieceUpdateHolder holder, RequestContext re
 
   private Future<Void> nonPackagePoLineUpdateInventory(PieceUpdateHolder holder, RequestContext requestContext) {
     return nonPackageUpdateTitleWithInstance(holder, requestContext)
-      .onSuccess(holder::withInstanceId)
+      .map(holder::withInstanceId)
       .compose(instanceId -> handleHolding(holder, requestContext))
```
Comment (damien-git) [inline src/main/java/org/folio/service/pieces/flows/update/PieceUpdateFlowInventoryManager.java:56]: `withInstanceId` returns a `PieceUpdateHolder`

File: src/main/java/org/folio/service/pieces/flows/update/PieceUpdateFlowInventoryManager.java [hunk 2]
```diff
@@ -64,10 +67,13 @@ private Future<Void> nonPackagePoLineUpdateInventory(PieceUpdateHolder holder, R
   private Future<Void> packagePoLineUpdateInventory(PieceUpdateHolder holder, RequestContext requestContext) {
     return titlesService.getTitleById(holder.getPieceToUpdate().getTitleId(), requestContext)
       .compose(title -> packageUpdateTitleWithInstance(title, requestContext))
-      .onSuccess(title -> holder.withInstanceId(title.getInstanceId()))
+      .map(title -> holder.withInstanceId(title.getInstanceId()))
       .compose(title -> handleHolding(holder, requestContext))
```
Comment (damien-git) [inline src/main/java/org/folio/service/pieces/flows/update/PieceUpdateFlowInventoryManager.java:71]: same here

**Review by damien-git — CHANGES_REQUESTED** [review]

**Comment by SerhiiNosko** [comment]
@damien-git naming has been improved per your suggestions, please re-review again

**Comment by damien-git** [comment]
> @damien-git naming has been improved per your suggestions, please re-review again

Thanks. You missed the first one.

**Review by damien-git — APPROVED** [review]

**Review by yuntianhu — APPROVED** [review]

**Comment by SerhiiNosko** [comment]
Karate tests passed after the changes:
image

**Pull request merged**